src/app/(admin)/admin/_lib/validationSchemas/specialistCommonSchemas.js

```diff
@@ -35,6 +36,50 @@ export const zInteger = z
 
 export const zUrl = zString.url({ message: MESSAGES.unacceptableValue });
 
+export const zWorkTimeSchema = z
+  .array(
+    z
+      .object({
+        weekDay: z.enum(Object.values(WEEKDAYS_TRANSLATION)),
+        time: z
+          .string()
+          .refine(val => !val || /\d{2}:\d{2}\s-\s\d{2}:\d{2}/.test(val), {
+            message: 'Введіть час у форматі ХХ:ХХ - ХХ:ХХ',
+          })
+          .nullish(),
+        isDayOff: z.boolean().nullish(),
+      })
+      .superRefine((data, ctx) => {
+        const { time, isDayOff } = data;
+        if (time && isDayOff) {
+          ctx.addIssue({
+            code: 'custom',
+            message: 'Приберіть час роботи, якщо це вихідний',
+            path: ['time'],
+          });
+        }
+      }),
+  )
+  .superRefine((workTime, ctx) => {
+    const specified = isSpecifiedWorkTime(workTime);
+    if (specified) {
+      workTime.forEach((day, index) => {
+        if (!day.isDayOff && !day.time) {
+          ctx.addIssue({
+            code: 'custom',
+            message: 'Додайте час роботи, якщо це не вихідний',
+            path: [index, 'time'],
+          });
+          ctx.addIssue({
+            code: 'custom',
+            message: `Додайте час роботи або вихідний для кожного дня
+            (чи приберіть усі дані якщо графік роботи не зазначено)`,
+          });
+        }
+      });
```

Comment:
It seems there's also [addIssues method](https://github.com/colinhacks/zod/blob/02699107616f672d79d3d71557456ec55597a4fa/deno/lib/ZodError.ts#L290C3-L290C12) that accepts array. If so this could be rewritten in a more functional way:

```javascript
  ctx.addIssues(
    workTime
      .filter(day => !day.isDayOff && !day.time)
      .map((day, index) => ([
          { code: 'custom', message: 'Додайте час роботи, якщо це не вихідний', path: [index, 'time'] },
          { code: 'custom', message: `Додайте час роботи або вихідний для кожного дня (чи приберіть усі дані якщо графік роботи не зазначено)` }
      ]))
    )
```

Reply:
Oh, OK (